src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -398,6 +420,15 @@
         (time! (get-metric :other-clean-time)
                (garbage-collect! db))
         (counters/inc! (get-metric :other-cleans)))
+      (when (request "gc_catalogs")
```

Comment:
Trying to understand this comment, maybe that should be updated, or the `other` condition should be moved after these two?:
```
;; It's important that this go last to ensure anything referencing
;; an env or resource param is purged first.
```